prime-router/src/main/kotlin/cli/tests/BasicTests.kt

```diff
@@ -115,126 +131,214 @@ class End2EndUniversalPipeline : CoolTest() {
+        var overallStatus = ""
 
-            val routeResults = pollForStepResult(convertReportId, TaskAction.route)
-            // verify each result is valid
-            for (result in routeResults.values)
-                passed = passed && examineStepResponse(result, "route", sender.topic)
-            if (!passed) {
-                bad("***async end2end_up FAILED***: Route result invalid")
-            }
-            val receiverNames = routeResults.values.flatMap { it?.reports ?: emptyList() }
-                .map { "${it.receivingOrg}.${it.receivingOrgSvc}" }
-            expectedReceivers.forEach { receiver ->
-                if (!receiverNames.contains(receiver.fullName)) {
-                    bad("***async end2end_up FAILED***: expected ${receiver.fullName} to have received a report")
-                    passed = false
-                }
+        echo("\nPolling for history endpoint for report ID: $reportId. (Max poll time $maxPollSecs seconds)")
+        while (timeElapsedSecs <= maxPollSecs) {
+            delay(pollSleepSecs.toLong() * 1000)
+            timeElapsedSecs += pollSleepSecs
+
+            val headers = mutableListOf<Pair<String, String>>()
+            val oktaToken = OktaCommand.fetchAccessToken(OktaCommand.OktaApp.DH_STAGE)
+            headers.add("Authorization" to "Bearer $oktaToken")
+            val getUrl = "${environment.url}/api/waters/report/$reportId/history"
+
+            val (_, response) = HttpClientUtils.getWithStringResponse(
+                url = getUrl,
+                accessToken = oktaToken,
+                timeout = 45000
+            )
+
+            overallStatus = JSONObject(response).getString("overallStatus")
+
+            if (DetailedSubmissionHistory.Status.DELIVERED.toString() == overallStatus) {
+                good("Report $reportId status was $overallStatus after $timeElapsedSecs seconds")
+                echo(response)
+                return response
             }
+        }
+
+        bad(
+            "***$name test FAILED***: Batch step polling for report $reportId failed. " +
+                "After $timeElapsedSecs seconds the report status is $overallStatus."
+        )
+        return ""
+    }
 
-            // check translate step
-            val routeReportIds = getAllChildrenReportId(convertReportId)
-            if (routeReportIds.size < expectedReceivers.size) {
-                return bad(
-                    "***async end2end_up FAILED***: Expected at least ${expectedReceivers.size} route" +
-                        "report id(s), but got ${routeReportIds.size}."
+    /**
+     * Runs validations after a report has been submitted to the UP and marked as Delivered.
+     * For all expected receivers this will validate:
+     * - An external filename matching that receiver is present in the history endpoint response
+     * - Download the file from BlobStorage
+     * - Compare the expected data with the actual data returned from BlobStorage
```

Comment:
We should also test that the name of the downloaded file is what we expect (I think there is a bug and you will find it's not what we expect?)